demos/smartlab_demo/python/object_detection/detector.py

```diff
@@ -0,0 +1,214 @@
+        #     "ruler"  , 1; "rider"  ,    omit; "scale"   , 1;
+        #     "hand", 2;
+        #  other conditions:
+        #     conf 0.2; nms 0.3
+        self.front1_exp = MwGlobalExp(
+            num_classes = 10,
+            fp_model = fp_front_models[0],
+            conf_thresh= 0.2,
+            nms_thresh = 0.3,
+            ie=ie,
+            device=device)
+
+        ###           frontview.global_subdetector2          ###
+        #  max-number constraints:
+        #     "weights", 6; "tweezers", 1; "battery", 1;
+        #  other conditions:
+        #     conf 0.1; nms 0.2
+        self.front2_exp = MwGlobalExp(
+            num_classes = 3,
+            fp_model = fp_front_models[1],
+            conf_thresh= 0.1,
+            nms_thresh = 0.2,
+            ie=ie,
+            device=device)
+
+        ### concatenate list of class names for topview
+        cls1 = self.top1_exp.mw_classes
+        cls2 = self.top2_exp.mw_classes
+        self.classes = cls1 + cls2
+        self.offset_cls_idx = [0, len(self.top1_exp.mw_classes)]
+        #  max-number constraints:
+        self.max_nums = {
+            "balance": 1, "weights": 6, "tweezers": 1,
+            "box"    : 1, "battery": 1, "tray"    : 2,
+            "ruler"  : 1, "rider"  : 1, "scale"   : 1,
+            "hand": 2,
+        }
+
+        ### build map of overlap classes
+        self.cls2tocls1 = {}
+        cls1_c2i = {c: i for i, c in enumerate(cls1)}
+        cls2_c2i = {c: (i + self.offset_cls_idx[1]) for i, c in enumerate(cls2)}
+        for cls in cls2:
+            if cls in cls1:
+                idx1 = cls1_c2i[cls]
+                idx2 = cls2_c2i[cls]
+                self.cls2tocls1[idx2] = idx1
+        self.repeat_cls2_ids = self.cls2tocls1.keys()
+        self.norepeat_num_classes = len(self.classes) - len(self.repeat_cls2_ids)
+
+        ### load models for top view
+        self.top1_subdetector = SubDetector(self.top1_exp, self.backend)
+        self.top2_subdetector = SubDetector(self.top2_exp, self.backend)
+        ### load models for front view
+        self.front1_subdetector = SubDetector(self.front1_exp, self.backend)
+        self.front2_subdetector = SubDetector(self.front2_exp, self.backend)
+
+    def _detect_one(self, img, view='top'):
+        if view == 'top': # top view
+            sub_detector1 = self.top1_subdetector
+            sub_detector2 = self.top2_subdetector
+        else: # front view
+            sub_detector1 = self.front1_subdetector
+            sub_detector2 = self.front2_subdetector
+
+        all_preds = []
+        for i, sub_detector in enumerate([sub_detector1, sub_detector2]):
+            outputs, img_info = sub_detector.inference(img)
+            if outputs[0] is not None:
+                preds = outputs[0] # work if bsize = 1
+            else:
+                continue
+            preds[:, 6] += self.offset_cls_idx[i]
+
+            if i == 0:
+                all_preds
```

Comment:
You just name the var without doing anything. Please remove this case

Reply:
wrongly deleted line.

fixed